common/cuckoo/common/elastic.py

```diff
@@ -90,15 +90,38 @@ def configure(self, hosts, analyses_index, tasks_index, events_index,
 
         self._max_result_window = max_result_window
         self._hosts = hosts
-        self._client = Elasticsearch(hosts, timeout=timeout,
-                                     http_auth=(user, password), ca_certs=ca_certs)
+        if ca_certs:
+            self._client = Elasticsearch(hosts, timeout=timeout,
+                                         http_auth=(user, password), ca_certs=ca_certs)
+        else:
+            self._client = Elasticsearch(hosts, timeout=timeout,
+                                         http_auth=(user, password))
         self._initialized = True
 
     def verify(self):
-        if not self.client.ping():
+        log.debug("Connecting to Elasticsearch.")
+        if self.client.ping():
+            verified = True
+            failures = 0
+        else:
+            verified = False
+            failures = 1
+
+        while not verified and failures < 6:
+            time.sleep(10)
+            if not self.client.ping():
+                failures += 1
+            else:
+                verified = True
+
+        if not verified:
```

Comment:
We already discussed this privately, but I'll try to re implement this using `async` methods.